Index: bloboats-1.0.1/Makefile
===================================================================
--- bloboats-1.0.1.orig/Makefile	2006-08-20 21:23:06.000000000 +0200
+++ bloboats-1.0.1/Makefile	2006-08-20 21:43:35.000000000 +0200
@@ -7,17 +7,19 @@
 ### Installation directory (Installation prefix)
 # If you don't have root privileges, set this to /home/tentacleman/bloboats/
 # for example, otherwise /. Don't use ~ or other environment variables!
-PREFIX = /
+PREFIX ?= /
+
+### Destination directory (Packager only)
+DESTDIR ?=
 
 ### Or if you don't want to run make install at all, uncomments the next line
 # and comment the line below. Compile and run with './bin/bloboats'
 #DATADIR = data
-
-DATADIR = $(PREFIX)/usr/games/bloboats/data
+DATADIR ?= $(PREFIX)/usr/games/bloboats/data
 
 ### If you do make install, you may perhaps want to set these
-BINARYDIR = $(PREFIX)/usr/bin
-CONFIGDIR = $(PREFIX)/etc
+BINARYDIR ?= $(PREFIX)/usr/bin
+CONFIGDIR ?= $(PREFIX)/etc
 
 PRIVATEDIR = ~/.bloboats
 
@@ -27,10 +29,10 @@
 # SDL, SDL_image, SDL_mixer and SDL_net to compile. Libvorbis for music support.
 # OpenGL support.
 
-SOUND = 1
-NETWORK = 1
-MUSIC = 1
-DEBUG = 0
+SOUND ?= 1
+NETWORK ?= 1
+MUSIC ?= 1
+DEBUG ?= 0
 
 # Select target system
 # Possible targets: DEFAULT (Linux and most of others too)
@@ -41,7 +43,7 @@
 
 
 #This should probably be DEFAULT
-TARGET = DEFAULT
+TARGET ?= DEFAULT
 
 
 ### Only some random settings below
@@ -133,17 +135,20 @@
 CXXFLAGS_CROSS_ = -g `$(SDLCONFIG) --cflags` -o bin/$(BINARY_) -O2
 CXXFLAGS_ = $(CXXFLAGS_$(TARGET)_)
 
-all: config main
+all: config-stamp build-stamp
 
-main: $(OBJ)
+build-stamp: build
+build: config-stamp $(OBJ)
 	$(CXX) $(OBJ) $(LIBS) $(CXXFLAGS)
 	$(STRIP) bin/$(BINARY)
+	touch build-stamp
 
 icon:
 	$(ICON_CMD)
 	$(CXX) -o bin/$(BINARY) $(OBJ) $(ICON) $(LIBS)
 	$(STRIP) bin/$(BINARY)
 
+config-stamp: config
 config:
 	echo "datadir $(DATADIR)" >bloboats.dirs
 	echo "privatedir $(PRIVATEDIR)" >>bloboats.dirs
@@ -153,16 +158,16 @@
 	echo "#define NETWORK $(NETWORK)" >> src/compiling_settings.h
 	echo "#define MUSIC $(MUSIC)" >> src/compiling_settings.h
 	echo "#define DEBUG $(DEBUG)" >> src/compiling_settings.h
+	touch config-stamp
 
-install:
-	mkdir -p $(BINARYDIR)/
-	mkdir -p $(CONFIGDIR)/
-	cp bloboats.dirs $(CONFIGDIR)/
-	mkdir -p $(DATADIR)/
-	cp -R data/* $(DATADIR)/
-	cp bin/$(BINARY) $(BINARYDIR)/
-	chmod -R 744 $(DATADIR)
-	chmod 755 `find $(DATADIR) -type d`
+install: build-stamp
+	mkdir -p $(DESTDIR)/$(BINARYDIR)/
+	mkdir -p $(DESTDIR)/$(CONFIGDIR)/
+	cp bloboats.dirs $(DESTDIR)/$(CONFIGDIR)/
+	mkdir -p $(DESTDIR)/$(DATADIR)/
+	cp -R data/* $(DESTDIR)/$(DATADIR)/
+	cp bin/$(BINARY) $(DESTDIR)/$(BINARYDIR)/
+	chmod 755 `find $(DESTDIR)/$(DATADIR) -type d`
 
 uninstall:
 	rm -f $(CONFIGDIR)/bloboats.dirs
@@ -177,7 +182,8 @@
 	$(STRIP) bin/$(BINARY_)
 
 clean:
-	rm bloboats.dirs
-	rm src/*.o
-	rm src/models/*.o
-	rm bin/$(BINARY)
+	rm -f bloboats.dirs
+	rm -f src/*.o
+	rm -f src/models/*.o
+	rm -f bin/$(BINARY)
+	rm -f *-stamp
