From: Debian Games Team <pkg-games-devel@lists.alioth.debian.org>
Date: Sun, 10 Jul 2016 22:12:47 +0200
Subject: uint32

---
 src/window.cpp | 2 +-
 src/window.h   | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/window.cpp b/src/window.cpp
index 8570465..78424b6 100644
--- a/src/window.cpp
+++ b/src/window.cpp
@@ -36,7 +36,7 @@ window::~window(){}
 
 
 
-SDL_Surface * window::OpenWindow(int w, int h, int b, int flags) {
+SDL_Surface * window::OpenWindow(int w, int h, int b, Uint32 flags) {
 	if (!flags) {
 		flags = oldflags^SDL_FULLSCREEN;
 	}
diff --git a/src/window.h b/src/window.h
index d17cbcb..9946884 100644
--- a/src/window.h
+++ b/src/window.h
@@ -31,7 +31,7 @@ class window
 		window();
 		~window();
 
-		SDL_Surface * OpenWindow(int width, int height, int bpp, int flags);
+		SDL_Surface * OpenWindow(int width, int height, int bpp, Uint32 flags);
 		void SetTitle(char *text, char *icon);
 		bool Iconify();
 
@@ -52,7 +52,7 @@ class window
 
 		int xlast, ylast;
 
-		int oldflags;
+		Uint32 oldflags;
 
 	private:
 		SDL_Surface *screen;
