Description: Fix segfault at start up
Author: Raphael Geissert <atomo64@gmail.com>
Bug-Debian: http://bugs.debian.org/523271

Index: bloboats-1.0.2.dsfg/src/window.cpp
===================================================================
--- bloboats-1.0.2.dsfg.orig/src/window.cpp	2012-03-06 20:16:41.778880814 +0700
+++ bloboats-1.0.2.dsfg/src/window.cpp	2012-03-06 20:18:06.671301767 +0700
@@ -36,7 +36,7 @@
 
 
 
-SDL_Surface * window::OpenWindow(int w, int h, int b, int flags) {
+SDL_Surface * window::OpenWindow(int w, int h, int b, Uint32 flags) {
 	if (!flags) {
 		flags = oldflags^SDL_FULLSCREEN;
 	}
Index: bloboats-1.0.2.dsfg/src/window.h
===================================================================
--- bloboats-1.0.2.dsfg.orig/src/window.h	2006-08-03 05:05:36.000000000 +0700
+++ bloboats-1.0.2.dsfg/src/window.h	2012-03-06 20:18:34.023437399 +0700
@@ -31,7 +31,7 @@
 		window();
 		~window();
 
-		SDL_Surface * OpenWindow(int width, int height, int bpp, int flags);
+		SDL_Surface * OpenWindow(int width, int height, int bpp, Uint32 flags);
 		void SetTitle(char *text, char *icon);
 		bool Iconify();
 
@@ -52,7 +52,7 @@
 
 		int xlast, ylast;
 
-		int oldflags;
+		Uint32 oldflags;
 
 	private:
 		SDL_Surface *screen;
