From: Markus Koschany <apo@debian.org>
Date: Sun, 10 Jul 2016 22:13:21 +0200
Subject: gcc-6

---
 Makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 879cc67..d6eab6a 100644
--- a/Makefile
+++ b/Makefile
@@ -60,10 +60,10 @@ ICON_CMD = i586-mingw32msvc-windres -i src/icon.rc -o $(ICON)
 
 
 ### C++ compiler to be used
-CXX_DEFAULT = g++
-CXX_FREEBSD = g++
-CXX_NETBSD = g++
-CXX_MACOSX = g++
+CXX_DEFAULT = g++-6
+CXX_FREEBSD = g++-6
+CXX_NETBSD = g++-6
+CXX_MACOSX = g++-6
 CXX_CROSS = i586-mingw32msvc-g++
 
 CXX = $(CXX_$(TARGET))
